explorer/src/providers/stats/solanaPerformanceInfo.tsx

```diff
@@ -0,0 +1,122 @@
+import { PerfSample } from "@solana/web3.js";
+import { ClusterStatsStatus } from "./solanaClusterStats";
+
+export type PerformanceInfo = {
+  status: ClusterStatsStatus;
+  avgTps: number;
+  historyMaxTps: number;
+  perfHistory: {
+    short: (number | null)[];
+    medium: (number | null)[];
+    long: (number | null)[];
+  };
+  transactionCount: number;
+};
+
+export enum PerformanceInfoActionType {
+  SetTransactionCount,
+  SetPerfSamples,
+  SetError,
+  Reset,
+}
+
+export type PerformanceInfoActionSetTransactionCount = {
+  type: PerformanceInfoActionType.SetTransactionCount;
+  data: number;
+};
+
+export type PerformanceInfoActionSetPerfSamples = {
+  type: PerformanceInfoActionType.SetPerfSamples;
+  data: PerfSample[];
+};
+
+export type PerformanceInfoActionSetError = {
+  type: PerformanceInfoActionType.SetError;
+  data: string;
+};
+
+export type PerformanceInfoActionReset = {
+  type: PerformanceInfoActionType.Reset;
+  data: PerformanceInfo;
+};
+
+export type PerformanceInfoAction =
+  | PerformanceInfoActionSetTransactionCount
+  | PerformanceInfoActionSetPerfSamples
+  | PerformanceInfoActionSetError
+  | PerformanceInfoActionReset;
+
+export function performanceInfoReducer(
+  state: PerformanceInfo,
+  action: PerformanceInfoAction
+) {
+  const status =
+    state.avgTps !== 0 && state.transactionCount !== 0
+      ? ClusterStatsStatus.Ready
+      : ClusterStatsStatus.Loading;
+
+  switch (action.type) {
+    case PerformanceInfoActionType.SetPerfSamples:
+      if (action.data.length < 1) {
+        return state;
+      }
+
+      let short = action.data.map((sample) => {
+        return sample.numTransactions / sample.samplePeriodSecs;
+      });
+
+      const historyMaxTps = Math.max(...short);
```

Comment:
I think this should be calculated from the final `short`, `medium`, and `long`. Otherwise if the max value happened over 30min ago, it will get downsampled away and the chart max will be exaggerated

Reply:
This actually happens before the short array is resized (so we're dealing with I believe 360 samples). Should be good.

Reply:
Right, I'm saying you should check for the max after the short array is resized. The max is used to determine the chart y axis and your code can pick up a max TPS that won't be charted because it's in the past / will not be visible after downsampling

Reply:
Oh, I misunderstood! Sure thing.

Reply:
Wait, I'm not sure we're on the same page here. The max TPS is being derived from the whole 6 hour history (before all the downsampling, it's 360 samples 6 * 60). Since I'm grabbing the max TPS from largest possible window (6h), this shouldn't get downsampled away. It should represent the highest value over the final short, medium, and long arrays.

Reply:
Okay, I do see one problem. The average of a downsampled max could be lower than the original max TPS. That might be your point lol. I will pull this from the short, medium, long arrays